espflash/src/targets/esp32c6.rs

```diff
@@ -113,10 +113,6 @@ impl Target for Esp32c6 {
     }
 
     fn supported_build_targets(&self) -> &[&str] {
-        &[
-            "riscv32imac-unknown-none-elf",
-            "riscv32imc-esp-espidf",
-            "riscv32imc-unknown-none-elf",
-        ]
+        &["riscv32imac-unknown-none-elf", "riscv32imac-esp-espidf"]
```

Comment:
Is `riscv32imc` really "invalid"? I'm not sure why you'd want to build without the atomic extension but it's perfectly valid to do so. I don't really understand the motivation behind removing it.

If we are removing it, then the ESP32-H2 needs to be updated as well.

Reply:
Not invalid, but almost certainly a mistake. Tbh, I've never really liked that we hard error here on a target mismatch. I think it should be a warning instead, but we can discuss that another time :D.

Reply:
For IDF, the `std` launches a huge linker error trying to find the functions for "A" for `std` unsuccessfully. So there, it is "invalid". I'll try `imc` for `no_std`, but I believe It'll fail for the same reason but `core` this time instead of `std`.

Reply:
I tested the `riscv32imc` on the C6. If you don't try to use `core::sync::atomic`, it compiles.

But, and this is my opinion: This is a new release of hardware, where among other things, the nice thing is the atomic support and no backward compatibility breaks by changing the target to `imac`. So, if you prefer to support `riscv32imac-unknown-none-elf` and `riscv32imc-unknown-none-elf`, let me know to add it too.

On the other hand, if you're already deciding to migrate to this hardware, and that "forces" you to change to `imac` the only work you have to do is move out from polyfill, and most of the embedded libs out there already have automatic detection and start using native support if it's available.

But for sure, if you try to use `riscv32imc-esp-espidf`, it doesn't compile. I opened an issue with the whole [detail](https://github.com/esp-rs/esp-idf-sys/issues/200)

Reply:
Sorry, for whatever reason GitHub doesn't let you control which lines are shown in the preview (kind of defeats the purpose, doesn't it?).

I was referring only to the bare-metal targets; I don't use the IDF stuff and frankly don't really care about it, you can do whatever you want with those targets.

Regardless, my opinion seems to be controversial for whatever reason, so as I requested earlier can we just update the H2 as well please so we can get this wrapped up?